### DISCUSSION TITLE
Release schedule/versioning?

### DISCUSSION THREAD

**Discussion by CanadaHonk** [facebook/hermes#1073] [General]
Is there any docs/info on a Hermes release schedule/versioning scheme/etc? It doesn't look like there is anything specific afaik (thought it might be tied with RN but don't think so?) Thanks!

**Comment by neildhar** [comment]
React Native currently cuts a Hermes release from trunk whenever a new release of RN is cut. This is necessary to maintain ABI compatibility with RN and to make it straightforward to patch bugs identified during the RN release process.

We also independently cut releases of the Hermes CLI, but we don't have a fixed cadence for that.

That said we're currently working on building a stable C-based ABI and developing a release process for Hermes. That could decouple the Hermes releases from RN, making it possible to swap out the Hermes binary for a more recent version without necessarily upgrading RN.

**Comment by tmikov** [comment]
Adding to what @neildhar said, ideally we would like Hermes release versions to be more meaningful and more predictable. For example, I personally like a scheme where the version contains the release year and month, and there is a six month release cadence.

At the same time we have to be careful to avoid confusion with RN release versions.

**Comment by tmikov** [comment]
The current plan:
1. Stable ABI so it is possible to drop future and even past version of Hermes when using a given RN release.
2. Hermes has a six month release cadence, either with increasing numbers or with year-month. 
- The exact format of the version is TBD, ideas welcome. 
- We must figure out how it would work with semver. 
- The latest release is labelled as `hermes-stable` and only receives bugfixes. The next release is `hermes-next`, where development happens.
3. RN just picks the current `hermes-stable` for their release.
